java/fory-core/src/test/java/org/apache/fory/ThreadSafeForyTest.java

```diff
@@ -373,4 +374,63 @@ public void testSerializerRegister() {
           return null;
         });
   }
+
+  @Test
+  public void testGetDepth() {
+    Fory fory = Fory.builder().requireClassRegistration(false).build();
+    Assert.assertEquals(fory.getDepth(), -1);
+    BeanA beanA = BeanA.createBeanA(2);
+    byte[] bytes = fory.serialize(beanA);
+    Assert.assertEquals(fory.getDepth(), -1);
+    fory.deserialize(bytes);
+    Assert.assertEquals(fory.getDepth(), -1);
+  }
+
+  @Test
+  public void testRegisterAfterSerializeThrowsException() throws Exception {
```

Comment:
Seems this test doesn't capture the bug in set `depth` to `-1`, please update this test to make it actually capture such bugs